modules/4_detection/src/detector.cpp

```diff
@@ -22,15 +42,90 @@ void Detector::detect(const cv::Mat& image,
                       std::vector<cv::Rect>& boxes,
                       std::vector<float>& probabilities,
                       std::vector<unsigned>& classes) {
-    CV_Error(Error::StsNotImplemented, "detect");
+    // Create 4D blob from BGR image
+    Blob::Ptr input = wrapMatToBlob(image);
+
+    req.SetBlob("image", input);
+
+    // Launch network
+    req.Infer();
+
+    float* output = req.GetBlob(outputName)->buffer();
+
+    int detectedRects = req.GetBlob(outputName)->size() / 7;
+
+    int xmin, ymin, xmax, ymax;
+
+    for (int i = 0; i < detectedRects; i++)
+    {
+        int tempRectIndex = i * 7;
+        float probability = output[tempRectIndex + 2];
+
+        if (probability > probThreshold)
+        {
+            int height = image.rows;
+            int width = image.cols;
+
+            xmin = output[tempRectIndex + 3] * width;
+            ymin = output[tempRectIndex + 4] * height;
+            xmax = output[tempRectIndex + 5] * width;
+            ymax = output[tempRectIndex + 6] * height;
+
+            Rect approvedRect(xmin, ymin, xmax - xmin, ymax - ymin);
```

Comment:
Oh no, I know what a problem. Please revert strict boxes check in the test and compute width and height as `xmax - xmin + 1` because if xmin == xmax that should be 1 pixel width (same for height)